packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -19,12 +19,23 @@ import '../globals.dart' as globals;
 import '../migrations/cmake_custom_command_migration.dart';
 import 'visual_studio.dart';
 
+/// These characters appear to be fine: @%()-+_{}[]`~
+const String badCharacters = r"'#!$^&*=|,;<>?";
+
 /// Builds the Windows project using msbuild.
 Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
   String? target,
   VisualStudio? visualStudioOverride,
   SizeAnalyzer? sizeAnalyzer,
 }) async {
+  final String projectPath = windowsProject.parent.directory.absolute.path;
+  final bool badPath = badCharacters.runes
+      .any((int i) => projectPath.contains(String.fromCharCode(i)));
+  if (badPath) {
+    throwToolExit(
+        'Path $projectPath contains invalid characters in "$badCharacters"');
```

Comment:
Is there some documentation and/or more context we can give the user about how to fix? A lot of times users will see a crash and then immediately copy paste it and file a bug asking for help.